ui/components/component-library/button-secondary/button-secondary.scss

```diff
@@ -20,7 +22,9 @@
     &:hover {
       color: var(--color-error-inverse);
       background-color: var(--color-error-default);
-      box-shadow: var(--component-button-danger-shadow);
+      box-shadow:
+        var(--shadow-size-sm)
+        var(--color-error-shadow);
     }
```

Comment:
### Before

Shadows for the `ButtonSecondary` component work as expected

### Before

https://github.com/MetaMask/metamask-extension/assets/8112138/0af684f1-93c1-4919-9536-826f4b89c64d

### After


https://github.com/MetaMask/metamask-extension/assets/8112138/0af6045b-3bed-4cbc-8d66-4b7c3913d1c9